core/src/main/java/io/grpc/internal/RetriableStream.java

```diff
@@ -642,6 +644,20 @@ public final Attributes getAttributes() {
     return Attributes.EMPTY;
   }
 
+  @Override
+  public void appendTimeoutInsight(InsightBuilder insight) {
+    synchronized (lock) {
+      insight.appendKeyValue("finished", closedSubstreamsInsight);
+    }
+    InsightBuilder drainedSubstreamsInsight = new InsightBuilder();
+    for (Substream sub : state.drainedSubstreams) {
```

Comment:
I don't think `drainedSubstreams` provide useful information, a previously failed attempt could be either drained or not-yet-drained, and a drained message could be either be buffered in DelayedStream or in RealStream, there is not much difference from the user's point of view, it only marks a progress of internal implementation.